CRM/MembershipExtras/Page/EditContributionRecurLineItems.php

```diff
@@ -302,21 +334,16 @@ private function calculateCurrentPeriodEndDate() {
    *
    * @throws \Exception
    */
-  private function getEarliestMembershipStartDate($lineItems) {
+  private function getEarliestLineStartDate($lineItems) {
     $earliestDate = null;
 
     foreach ($lineItems as $line) {
-      if ($line['entity_table'] != 'civicrm_membership') {
-        continue;
-      }
-
-      $membership = $this->getMembership($line['entity_id']);
-      $membershipStartDate = new DateTime($membership['start_date']);
+      $startDate = new DateTime($line['start_date']);
```

Comment:
shouldn't you only check the membership line items  ?

Reply:
Before when we only looked at  membership start dates it made sense, but you could, for example, acquire a membership for only the first year and not renew it, but keep paying a donation, or an event in subsequent years. What start date would we use then, if there is no membership? This would make it work on those cases.